src/formats.c

```diff
@@ -1621,7 +1621,7 @@ char *fmt_self_test(struct fmt_main *format, struct db_main *db)
 	self_test_running = 0;
 
 	MEM_FREE(salt_alloc);
-	BLOB_FREE(binary_alloc);
+	//BLOB_FREE(binary_alloc);
```

Comment:
I can't yet see why this fails with ASan unless commented out. It boils down to this after expansion:
```c
	if (binary_alloc && (format->params.flags & FMT_BLOB) &&
		(((fmt_data*)binary_alloc)->flags & FMT_DATA_ALLOC)) 
			MEM_FREE(((fmt_data*)binary_alloc)->blob);
```

Reply:
...Oh and as we all know, `MEM_FREE` boils down to checking that `((fmt_data*)binary_alloc)->blob` is not NULL either. So why the heck does it say BOOM with
```
==49736==The signal is caused by a WRITE memory access.
==49736==Hint: address points to the zero page.
```